collect_app/src/main/java/org/odk/collect/android/database/instances/InstanceDatabaseMigrator.java

```diff
@@ -143,6 +150,14 @@ private void upgradeToVersion8(SQLiteDatabase db) {
         db.execSQL("UPDATE " + INSTANCES_TABLE_NAME + " SET " + CAN_DELETE_BEFORE_SEND + " = 'true';");
     }
 
+    private void upgradeToVersion9(SQLiteDatabase db) {
+        String temporaryTable = INSTANCES_TABLE_NAME + "_tmp";
```

Comment:
I think we can just add the column given that the default value for `editOf` will be `null` right? Can we not use `ALTER TABLE ADD COLUMN` here?

Reply:
We could if it was just that, but we also change the table definition by adding the foreign key, and this is not possible without recreating the table.

Reply:
It looks to me like you can use the "shorthand" syntax (mentioned annoyingly offhand in the docs):

```sql
ALTER TABLE instances ADD COLUMN editOf INTEGER REFERENCES instances(_id)
```

I played around with removing `editOf` from the `CREATE TABLE` and adding it with the above, and we still get test passes for the constraint. We could also use this syntax in the `CREATE TABLE` now I think about it as it's a bit more contained - you don't end up with two lines setting the foreign key up.

Reply:
The [`ADD COLUMN` syntax](https://sqlite.org/lang_altertable.html) is (in simplified terms) `ADD COLUMN <columndef>` where the `columndef` is just `<name> <type> <constraint(s)>`. So as with a constraint like `NOT NULL` you can add a `CHECK`:

```sql
"ALTER TABLE instances ADD COLUMN editOf INTEGER REFERENCES instances(_id) CHECK (editOf != _id)"
```

This worked for me against our tests. 

By the way, I often find playing around with a SQLite GUI useful for working these kinds of expressions out. I've used https://sqlitebrowser.org/ in the past. You can even just pull a DB off an emulator device and then mess around with it locally.

Reply:
Right it works!